cynic-querygen/src/lib.rs

```diff
@@ -42,6 +41,21 @@ pub enum Error {
 
     #[error("an enum-like value was provided to an argument that is not an enum")]
     ArgumentNotEnum,
+
+    #[error("expected an input object, enum or scalar")]
+    ExpectedInputType,
+
+    #[error("expected an enum, scalar, object, union or interface")]
+    ExpectedOutputType,
+
+    #[error("expected a homogenous list of input values")]
+    ExpectedHomogenousList,
+
+    #[error("expected field to be a list")]
+    ExpectedListType,
+
+    #[error("expected a value that is or contains an input object")]
+    ExpectedInputObjectValue,
```

Comment:
Just testing some stuff:

````suggestion
    ```
    #[error("expected a value that is or contains an input object")]
    ExpectedInputObjectValue,
    ```
````